Pioneer AVR Binding:  Zone 2 Control

Has anyone looked at implementing Zone 2 control with the Pioneer AVR binding, or is it already possible?
##

It looks like the zone 2 volume control command is never getting through to the receiver.  I'm not getting anything in my pioneer.log file.  I am seeing this in my openhab.log:

```
2016-12-11 13:28:51.950 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occured while calling handler: org.openhab.binding.pioneeravr.protocol.AvrConnectionException: The parameter v does not match the pattern [0-9]{3}
 does not match the pattern [0-9]{3}ion: org.openhab.binding.pioneeravr.protocol.AvrConnectionException: The parameter value 57 of the command ZV
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)[:1.8.0_65]
        at java.util.concurrent.FutureTask.get(FutureTask.java:206)[:1.8.0_65]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:186)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:79)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:63)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:352)[103:org.eclipse.smarthome.core.thing:0.9.0.201612010841]
        at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:46)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:177)[96:org.eclipse.smarthome.core:0.9.0.201612010841]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
        at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
 does not match the pattern [0-9]{3}eravr.protocol.AvrConnectionException: The parameter value 57 of the command ZV
        at org.openhab.binding.pioneeravr.internal.protocol.ParameterizedCommand.getCommand(ParameterizedCommand.java:84)[223:org.openhab.binding.pioneeravr:2.0.0.201612101322]
        at org.openhab.binding.pioneeravr.internal.protocol.StreamAvrConnection.sendCommand(StreamAvrConnection.java:244)[223:org.openhab.binding.pioneeravr:2.0.0.201612101322]
        at org.openhab.binding.pioneeravr.internal.protocol.StreamAvrConnection.setVolume(StreamAvrConnection.java:403)[223:org.openhab.binding.pioneeravr:2.0.0.201612101322]
        at org.openhab.binding.pioneeravr.internal.protocol.StreamAvrConnection.sendVolumeCommand(StreamAvrConnection.java:373)[223:org.openhab.binding.pioneeravr:2.0.0.201612101322]
        at org.openhab.binding.pioneeravr.internal.handler.AbstractAvrHandler.handleCommand(AbstractAvrHandler.java:265)[223:org.openhab.binding.pioneeravr:2.0.0.201612101322]
        at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:355)[103:org.eclipse.smarthome.core.thing:0.9.0.201612010841]
        at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:1)[103:org.eclipse.smarthome.core.thing:0.9.0.201612010841]
        ... 5 more
```

@Stratehm is is possible to implement Zone2Control for the VSX-924?
I am on 2.1 stable. At the moment when triggering the Zone2 Power i get the folling log entry:

`2017-07-17 18:46:21.493 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occurred while calling handler: java.lang.ArrayIndexOutOfBoundsException: -1
java.util.concurrent.ExecutionException: java.lang.ArrayIndexOutOfBoundsException: -1
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:220)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:189)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:83)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:67)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:374)[105:org.eclipse.smarthome.core.thing:0.9.0.b5]
	at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:47)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:181)[98:org.eclipse.smarthome.core:0.9.0.b5]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]
Caused by: java.lang.ArrayIndexOutOfBoundsException: -1
	at org.openhab.binding.pioneeravr.internal.protocol.SimpleCommand$SimpleCommandType.getCommand(SimpleCommand.java:50)[209:org.openhab.binding.pioneeravr:2.1.0]
	at org.openhab.binding.pioneeravr.internal.protocol.SimpleCommand.getCommand(SimpleCommand.java:64)[209:org.openhab.binding.pioneeravr:2.1.0]
	at org.openhab.binding.pioneeravr.internal.protocol.StreamAvrConnection.sendCommand(StreamAvrConnection.java:151)[209:org.openhab.binding.pioneeravr:2.1.0]
	at org.openhab.binding.pioneeravr.internal.protocol.StreamAvrConnection.sendPowerCommand(StreamAvrConnection.java:217)[209:org.openhab.binding.pioneeravr:2.1.0]
	at org.openhab.binding.pioneeravr.internal.handler.AbstractAvrHandler.handleCommand(AbstractAvrHandler.java:167)[209:org.openhab.binding.pioneeravr:2.1.0]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$4.call(ThingManager.java:377)[105:org.eclipse.smarthome.core.thing:0.9.0.b5]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$4.call(ThingManager.java:1)[105:org.eclipse.smarthome.core.thing:0.9.0.b5]
	at org.eclipse.smarthome.core.common.SafeMethodCaller.executeDirectly(SafeMethodCaller.java:218)[98:org.eclipse.smarthome.core:0.9.0.b5]
	... 12 more`

--- a/addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/PioneerAvrBindingConstants.java
+++ b/addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/PioneerAvrBindingConstants.java
@@ -51,7 +51,7 @@ public class PioneerAvrBindingConstants {
     public static final String DISPLAY_INFORMATION_CHANNEL = "displayInformation#displayInformation";
 
     public static final String GROUP_CHANNEL_PATTERN = "zone%s#%s";
-    public static final Pattern GROUP_CHANNEL_ZONE_PATTERN = Pattern.compile("zone([0-1])#.*");
+    public static final Pattern GROUP_CHANNEL_ZONE_PATTERN = Pattern.compile("zone([0-3])#.*");
 
     // Used for Discovery service
     public static final String MANUFACTURER = "PIONEER";

--- a/addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/internal/protocol/Response.java
+++ b/addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/internal/protocol/Response.java
@@ -31,7 +31,7 @@ public class Response implements AvrResponse {
      */
     public enum ResponseType implements AvrResponse.RepsonseType {
         POWER_STATE("[0-2]", "PWR", "APR", "BPR"),
-        VOLUME_LEVEL("[0-9]{3}", "VOL", "ZV", "YV"),
+        VOLUME_LEVEL("[0-9]{2,3}", "VOL", "ZV", "YV"),
         MUTE_STATE("[0-1]", "MUT", "Z2MUT", "Z3MUT"),
         INPUT_SOURCE_CHANNEL("[0-9]{2}", "FN", "Z2F", "Z3F"),
         DISPLAY_INFORMATION("[0-9a-fA-F]{30}", "FL");

